crypto/ec/ecp_smpl.c

```diff
@@ -1363,3 +1365,57 @@ int ec_GFp_simple_field_sqr(const EC_GROUP *group, BIGNUM *r, const BIGNUM *a,
 {
     return BN_mod_sqr(r, a, group->field, ctx);
 }
+
+/*-
+ * Apply randomization of EC point projective coordinates:
+ *
+ *   (X, Y ,Z ) = (lambda^2*X, lambda^3*Y, lambda*Z)
+ *   lambda = [1,group->field)
+ *
+ */
+int ec_GFp_simple_blind_coordinates(const EC_GROUP *group, EC_POINT *p,
+                                    BN_CTX *ctx)
+{
+    int ret = 0;
+    BIGNUM *lambda = NULL;
+    BIGNUM *temp = NULL;
+
+    BN_CTX_start(ctx);
+    lambda = BN_CTX_get(ctx);
+    temp = BN_CTX_get(ctx);
+    if (temp == NULL) {
+        ECerr(EC_F_EC_GFP_SIMPLE_BLIND_COORDINATES, ERR_R_MALLOC_FAILURE);
+        goto err;
+    }
+
+    /* make sure lambda is not zero */
+    do {
+        if (!BN_rand_range(lambda, group->field)) {
```

Comment:
This should probably be BN_priv_rand_range()

Reply:
No problem in changing this, but I think this calls for further discussion: we had an internal conversation about it and we would like to have your (and @richsalz , @mattcaswell ) comments on the issue.

The [documentation](https://github.com/openssl/openssl/blob/master/doc/man3/RAND_bytes.pod) states:

> RAND_priv_bytes() has the same semantics as RAND_bytes(). It is intended to be used for generating long-term private keys. If using the default RAND_METHOD, this function uses a separate instance of the PRNG so that a compromise of the global generator will not affect such key generation.

(*ultimately the difference between `BN_rand_range()` and `BN_priv_rand_range()` is that the latter uses `RAND_priv_bytes()` rather than `RAND_bytes()`, which in turn, currently, is relevant only when using the DRBG-based default `RAND_METHOD` implementation*)

We are already *violating* the documentation in some places, e.g. by using `BN_priv_rand_range()` in nonce generation in ECDSA, which clearly isn't a long term private key, but can be justified as basically we are generating an ephemeral key, and we want it to mirror as closely as possible the long term key generation.

I think that if instead `RAND_priv_bytes()` is intended to be used for generating any sort of *private* random number, independently from its life length, this should be clarified in the documentation.

What are your thoughts?

Reply:
You are right, the documentation states the original intention but that has been shifted slightly from "long-term private keys" to "everything that is intended to remain private" by Kurt. Since Kurt did most of the changes and there haven't  been any objections to his point of view, it can probably be considered the current consensus. So yes, the documentation should be changed accordingly.

Reply:
Thanks @mspncp!

@kroeckx should I open a separate issue to update the documentation?

Reply:
fixed and squashed !

Reply:
@mspncp , @kroeckx , @richsalz I opened PR #6514 to fix the documentation issue.